src/nvim/message_defs.h

```diff
@@ -10,14 +10,25 @@ typedef struct {
 } HlMessageChunk;
 
 typedef kvec_t(HlMessageChunk) HlMessage;
+typedef int64_t MsgID;
 
+#define MSG_KIND_PROGRESS "progress"
+
+typedef struct msg_ext_data {
+  Integer percent;        ///< Progress percentage
+  String title;           ///< Title for progress message
+  String status;          ///< Status for progress message
+  DictOf(String, Object) extra_info;  ///< Extra info for 'echo' messages
+} MessageExtData;
```

Comment:
it's not related to "ext". it's just messages. also not sure why we are flip-flopping between "Message" and "Msg". Let's just stick with Msg, even though some old names use "Message".

```suggestion
} MsgData;
```